### PR TITLE
Login/Register Page E2E Tests

### PR DESCRIPTION
**Pull request by minsooerickim**
in progress

### CODE REVIEW

File: app/client/e2e/login.spec.ts
Comment (kadey001) [inline app/client/e2e/login.spec.ts]: Could you explain why the toHaveText is formatted like this and se
Screenshot from 2022-05-02 16-20-39
ems to have an invisible Unicode character?

File: app/client/e2e/login.spec.ts
Comment (kadey001) [inline app/client/e2e/login.spec.ts]: same here

File: app/client/e2e/login.spec.ts
Comment (minsooerickim) [reply app/client/e2e/login.spec.ts]: I was actually a bit confused about that as well. I initially did `tohaveText('Email')` because that was how it was set in our code; However, after running the test, playwright said it expected to see `'Email *Email *'` as the label on the email input field so I temporarily put that in my testing code to see if it even passes. I am in the progress of looking into other methods I could potentially use to go around this issue. (same thing for password field)

File: app/client/e2e/login.spec.ts
Comment (kadey001) [inline app/client/e2e/login.spec.ts]: There is no assertion in this test. You'd want to do an expect for that text to exist/the element to be visible.

File: app/client/e2e/login.spec.ts
Comment (kadey001) [inline app/client/e2e/login.spec.ts]: It might be a good idea to wrap the tests in describe blocks. You could break it up to things like `errors` `page` `redirects` for example. Could really break it up however you like as long as it makes sense. Not too important on this page since the tests are simple and there won't be very many, but it will be important on more complex pages.

**Review by kadey001 — CHANGES_REQUESTED** [review]

**Comment by kadey001** [comment]
Looking good. You also want to test that the login and register pages both succeed when putting in valid data.

**Comment by minsooerickim** [comment]
> Looking good. You also want to test that the login and register pages both succeed when putting in valid data.

for sure. I was also thinking of adding a feature to display an error if the user clicks login/register with empty fields because currently, the error messages show only when all the fields are filled out

File: app/e2e/tests/account/login.spec.ts
Comment (DevelopmentByDavid) [inline app/e2e/tests/account/login.spec.ts]: I think screenshot testing would be more ideal for this type of thing

https://playwright.dev/docs/screenshots
https://playwright.dev/docs/test-snapshots

File: app/e2e/tests/account/login.spec.ts
Comment (DevelopmentByDavid) [inline app/e2e/tests/account/login.spec.ts]: Why is there a tab press?

File: app/e2e/tests/account/register.spec.ts
Comment (DevelopmentByDavid) [inline app/e2e/tests/account/register.spec.ts]: snapshot testing imo, which will also capture things like alignment changing unintentionally and not just that content is present.

**Comment by DevelopmentByDavid** [comment]
Just an FYI that I'm working on reviewing this. Will probably finish the review tomorrow.

File: app/e2e/tests/account/register.spec.ts
Comment (DevelopmentByDavid) [inline app/e2e/tests/account/register.spec.ts]: Why is this tab? Are we depending on the order of the form? Will the test break if we change the form's order?

File: app/e2e/package.json
Comment (DevelopmentByDavid) [inline app/e2e/package.json]: Is this a notable change that would need to be documented?

File: app/server/package.json
Comment (DevelopmentByDavid) [inline app/server/package.json]: I think the documentation needs to be updated?

File: app/server/prisma/seed.ts
Comment (DevelopmentByDavid) [inline app/server/prisma/seed.ts]: @kadey001 IMO some of this pretest stuff should be utility functions instead. It would be slightly repetitive but it would also give finer grained control over the tests as a whole and each environment for each test.

**Review by DevelopmentByDavid — CHANGES_REQUESTED** [review]
Looks good, some minor comments / changes.

File: app/e2e/tests/account/login.spec.ts
Comment (kadey001) [reply app/e2e/tests/account/login.spec.ts]: Tab is one way to get to the next entry in a form rather than clicking on the next box. I'll try and update it to be more consistent with clicking or tab but not a mix.

File: app/e2e/tests/account/register.spec.ts
Comment (kadey001) [reply app/e2e/tests/account/register.spec.ts]: Good point, it would break if you changed the order. I'll update it to just use clicking and no tabs.

File: app/e2e/package.json
Comment (kadey001) [reply app/e2e/package.json]: I don't think so, it was just a mistake when I originally set it up, it should have had the pretest there.

File: app/server/package.json
Comment (kadey001) [reply app/server/package.json]: I thought I did update it for this, but I can double check.

File: app/server/prisma/seed.ts
Comment (kadey001) [reply app/server/prisma/seed.ts]: I agree, as we discussed in the other PR we will update to using utility functions. I might wait until the other PR is merged since he is working on creating those utility functions and I don't want to create conflicts by also implementing them in this PR.

**Pull request merged**
